checks/evaluation/vulnerabilities.go

```diff
@@ -15,46 +15,40 @@
 package evaluation
 
 import (
-	"fmt"
-	"strings"
-
-	"github.com/google/osv-scanner/pkg/grouper"
-
 	"github.com/ossf/scorecard/v4/checker"
 	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/hasOSVVulnerabilities"
 )
 
 // Vulnerabilities applies the score policy for the Vulnerabilities check.
-func Vulnerabilities(name string, dl checker.DetailLogger,
-	r *checker.VulnerabilitiesData,
+func Vulnerabilities(name string,
+	findings []finding.Finding,
+	dl checker.DetailLogger,
 ) checker.CheckResult {
-	if r == nil {
-		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+	expectedProbes := []string{
+		hasOSVVulnerabilities.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
 		return checker.CreateRuntimeErrorResult(name, e)
 	}
 
-	aliasVulnerabilities := []grouper.IDAliases{}
-	for _, vuln := range r.Vulnerabilities {
-		aliasVulnerabilities = append(aliasVulnerabilities, grouper.IDAliases(vuln))
+	vulnsFound := 0
+	for _, f := range findings {
+		if f.Outcome == finding.OutcomeNegative {
+			// Log all the negative findings.
+			checker.LogFindings(negativeFindings(findings), dl)
```

Comment:
This will create a quadratic number of logs wont it?

For each negative finding
   log all negative findings

Reply:
the logging should take place out of the loop